### ISSUE TITLE
Release plan - v14.x Maintenance LTS

### ISSUE THREAD

**Issue by BethGriggs** [nodejs/Release#567]
_Draft schedule - all dates subject to change_

| Release candidate | Release Date | Release | Releaser |
|:-:| ------------- |:--------:| ---------|
| - | 2020-10-27     | v14.15.0   | @richardlau   *LTS Transition Release 
| - | 2020-11-16     | v14.15.1 (Security) | @BethGriggs 
|  2020-12-08  | 2020-12-15       | v14.15.2  | @BethGriggs
|  | 2020-12-17 | v14.15.3 | @BethGriggs 
| 2021-01-04 | 2021-01-04 | v14.15.4 (security) | @BethGriggs
|  | 2021-02-09 | v14.15.5 | @BethGriggs 
| - | 2021-02-23 | v14.16.0 | @BethGriggs 
| - | 2021-04-06  | v14.16.1  | @mylesborins 
| 2021-05-11 | 2021-05-11 | v14.17.0  | @danielleadams
| 2021-06-05  | 2021-06-15     | v14.17.1  | @targos
| -  | 2021-07-01   | v14.17.2 (Security) | @richardlau
| -  | 2021-07-05   | v14.17.3  | @richardlau
| 2021-07-20 | 2021-07-29 | v14.17.4 (Security) | @richardlau
| 2021-09-07  | 2021-09-28     | v14.18.0  | @targos
|  | 2021-10-12     | v14.18.1 (Security) | @danielleadams 
|  | 2021-11-30 | v14.18.2 | @richardlau 
| | 2022-01-10 | v14.18.3 (Security) | @richardlau 
| 2022-01-25 | 2022-02-01 | v14.19.0 | @richardlau 
| - | 2022-03-17 | v14.19.1 (Security) | @richardlau 
| 2022-04-X | 2022-05-04 | v14.19.2 | @juanarbol 
| | 2022-05-17 | v14.19.3 | @richardlau 
| | 2022-07-07 | v14.20.0 (Security) | @juanarbol (@danielleadams)
| | 2022-09-23 | v14.20.1 | @bengl 
| | 2022-11-01 | v14.21.0 | @danielleadams 
| | 2022-11-04 | v14.21.1 (Security) | @BethGriggs 
| | 2022-12-13 | v14.21.2 | @richardlau 
| | 2023-02-16 | v14.21.3 | @richardlau (Security)
| | 2023-03-xx | v14.21.4 | (Tentative last release before EOL)
<details>
<summary>Current</summary>

| Release candidate | Release Date | Release | Releaser |
|:-:| ------------- |:--------:| ---------|
| - | 2020-04-21    | v14.0.0   | @BethGriggs 
| - | 2020-04-28   | v14.1.0   | @BethGriggs
| - | 2020-05-05    | v14.2.0   |  @targos 
| - | 2020-05-19    | v14.3.0   | @codebytere
| - | 2020-06-02    | v14.4.0 (Security)  |  @targos
| - | 2020-06-04    | Skipped   |  -
| - | 2020-06-30    | v14.5.0   |  @codebytere
| - | 2020-07-21    | v14.6.0   |  @MylesBorins
| - | 2020-07-29    | v14.7.0   |  @ruyadorno / @MylesBorins 
| - | 2020-08-11     | v14.8.0   |  @codebytere 
| - | 2020-08-25    | v14.9.0  |  @danielleadams / @BethGriggs
| - | 2020-09-08    | v14.10.0   |  @richardlau 
| - | 2020-09-10    | v14.10.1   |  @richardlau 
| - | 2020-09-15    | v14.11.0 (Security) |  @richardlau 
| - | 2020-09-22    | v14.12.0   |  @ruyadorno
| - | 2020-10-06    | v14.13.0   |  @danielleadams / @BethGriggs
| - | 2020-10-15    | v14.14.0   | @MylesBorins 
</details>

**Comment by targos** [comment]
@BridgeAR since there's a security release today, I suggest to postpone the regular release by two days (or a week if that's more convenient for you)

**Comment by targos** [comment]
We missed the release last week. Do we want to do one tomorrow or wait for June 30?

**Comment by BethGriggs** [comment]
@targos I think doing one this week makes sense if someone has the availability to do so.

**Comment by codebytere** [comment]
Staging should be pretty updated, i audited most everything within the last few days - I’m ok with either but if we do one this week we might have next to nothing in the release on the 30th 🤔

**Comment by MylesBorins** [comment]
@ruyadorno and I are going to pair on the 7-28 release. 🎉

**Comment by MylesBorins** [comment]
As a heads up we had a number of issues with the 14.x release last week and are aiming to get it out on the 21st

**Comment by ruyadorno** [comment]
heads up, we're going to be pushing the 7-28 release a day, to Wednesday 29th instead 😊 /cc @MylesBorins

**Comment by codebytere** [comment]
Should we keep our original date of next tues for the next release? Given that the last one was only ~week ago it might make sense to push it out 🤔

**Comment by MylesBorins** [comment]
We usually have done 14.x every two weeks. I don't see a need to push it personally

**Comment by ruyadorno** [comment]
Signed up for the September 22nd release 😊

**Comment by MylesBorins** [comment]
There was a noticeable performance regression in 14.13.0 with the CJS exports change. Guy has landed a significant perf improvement over the weekend

https://github.com/nodejs/node/issues/35574
https://github.com/nodejs/node/pull/35583

Thinking we might want to push out another release this week. Thoughts?

**Comment by targos** [comment]
+1

**Comment by BethGriggs** [comment]
+1, assuming someone has availability to prepare the release

**Comment by cjihrig** [comment]
I'd like to see https://github.com/nodejs/node/pull/35602 land before the next release. The new `rm()` methods in `fs` don't currently handle input validation errors properly. For some reason those new APIs weren't introduced as experimental (as far as I can tell), so correcting the error handling would need to happen before the next release, or it will become a breaking change.

**Comment by bcoe** [comment]
If possible, I would love for us to land [this bug fix for rimraf](https://github.com/nodejs/node/pull/35566) too, which we found while working on the `rm()` feature.

It seems to address the known issue https://github.com/nodejs/node/issues/34266 ... and, in general, seems like it could be a pretty gnarly bug for people (_basically rimraf always runs at least twice, rather than once_).

_this is not a new bug, so it's not the end of the world if we miss the release window._

**Comment by BethGriggs** [comment]
On the release mentoring call, none of the releasers had availability this week. ping @nodejs/releasers does anyone have the availability to prepare a release this week?

(I am in the process of landing a few bits on `v14.x-staging`, but I will not have time to prepare the release.)

**Comment by MylesBorins** [comment]
I can potentially find time tomorrow to prepare / remote if staging is mostly ready to go

**Comment by MylesBorins** [comment]
Just finished going through the backlog with branchdiff. Testing locally and I'll get a release proposal up

**Comment by dilyanpalauzov** [comment]
Will the optional assignment be backported to v14?

**Comment by MylesBorins** [comment]
@dilyanpalauzov to the best of my knowledge optional assignment feature came from an update to V8, which I do not believe we will be able to land on 14

@targos can you please confirm

**Comment by targos** [comment]
It's unlikely that we're able to land the V8 update on 14, but I'm still going to try soon :)

**Comment by gabrielschulhof** [comment]
May we please have https://github.com/nodejs/node/pull/37217 in the next v14.x release? It should make backporting subsequent commits a lot easier.

**Comment by targos** [comment]
We shouldn't forget https://github.com/nodejs/node/issues?q=label%3Abaking-for-lts+is%3Aclosed while preparing the next semver-minor release

**Comment by rvphilip** [comment]
I noticed some experts say that this release marks the transition of Node.js 14.x into Long Term Support (LTS) with the codename 'Fermium'. Therefore the 14.x release line now will move into "Active LTS" and will remain so until October 2021. I hope this is true because the version that I'm running now supports the  Windows 64-bit Installer. 64 bit was also a trusted component of several [Enterprise Application Development Services](https://thinkpalm.com/services/enterprise-application-services/) and Node.js 14.x is expected to help them function more properly. I just wanted to know if the maintenance will last till 2023 or not? Cheers

**Comment by AlanGreene** [comment]
@rvphilip the release schedule (including LTS status) is documented at https://nodejs.org/en/about/releases/. According to that doc v14 (Fermium) entered active LTS on 2020-10-27, will enter maintenance LTS on 2021-10-19, and will be end-of-life on 2023-04-30.

**Comment by Trott** [comment]
Is it likely that there will be one final release of 14.x or are we likely done barring a catastrophic security issue or something like that?

**Comment by targos** [comment]
I think it's unlikely at this point, but is there something you would like to see land on v14.x?

**Comment by Trott** [comment]
> I think it's unlikely at this point, but is there something you would like to see land on v14.x?

Nope, I just wanted to make sure I was giving someone else accurate information. Thanks!

**Comment by richardlau** [comment]
There are currently 8 [lts-watch-v14.x ](https://github.com/nodejs/node/labels/lts-watch-v14.x) labelled PRs
https://github.com/nodejs/node/issues?q=is%3Aclosed+label%3Alts-watch-v14.x

I'm not sure anything there compels us to do a final 14.x release.
Other things that could go in a final 14.x release are a timezone update (2023c, needs manual PR targeting the version of ICU in v14.x-staging) and root certificate update (no PR yet, I have an automation PR open https://github.com/nodejs/node/pull/47425 which would open a PR on `main` to update the root certificates). But those will be untested (the timezone update might go out in 19.9.0 but that's for a different version of ICU).
